app/components/channel_drawer_list/channel_drawer_list.js

```diff
@@ -51,23 +65,40 @@ class ChannelDrawerList extends Component {
         this.firstUnreadChannel = null;
         this.lastUnreadChannel = null;
         this.state = {
+            dataSource: this.buildData(props),
+            searching: false,
             showAbove: false,
             showBelow: false,
-            dataSource: this.buildData(props)
+            term: null
         };
+
         MaterialIcon.getImageSource('close', 20, this.props.theme.sidebarHeaderTextColor).
         then((source) => {
             this.closeButton = source;
         });
     }
 
     shouldComponentUpdate(nextProps, nextState) {
-        return !deepEqual(this.props, nextProps, {strict: true}) || !deepEqual(this.state, nextState, {strict: true});
+        const currentProps = {...this.props};
+        const props = {...nextProps};
+
+        // remove the profiles, otherChannels, myPreferences and statuses
+        // to prevent the component to re-render every time they change
+        Reflect.deleteProperty(currentProps, 'profiles');
```

Comment:
Yeah, sounds good.. is going to be a lot of work, maybe we can create another ticket to split this into smaller components?